dist/BaseConfig.d.ts

```diff
@@ -46,6 +53,7 @@ export declare type ComponentConfig = Partial<{
     layers: Array<LayerSpec>;
     tabs: Array<TabSpec>;
     contextMenuItems: Array<ContextMenuItem>;
+    mouseInteractionHooks: MouseInteractions;
```

Comment:
this is very misleading, since its not a hook,
better name it: 'MouseInteractionCallbacks" or something like that,
and events should start with 'on' in typescript so please refactor them to
onMouseDown...
onMouseUp...